features/step_definitions/attachment_steps.rb

```diff
@@ -26,7 +26,11 @@
 end
 
 When(/^I add an external attachment with the title "(.*?)" and the URL "(.*?)"$/) do |title, url|
-  create_external_attachment(url, title)
+  create_external_attachment(url, title, nil)
+end
+
+When(/^I add an external attachment with the title "(.*?)" and the URL "(.*?)" with design system permission$/) do |title, url|
+  create_external_attachment(url, title, "design_system")
 end
```

Comment:
instead of a new step, we can just do this:

```
@user.has_permission? "Preview design system"
```

Reply:
Ah right, nice!